src/pc/pc_main.c

```diff
@@ -266,8 +266,17 @@ void main_func(void) {
 #endif
 }
 
+//special Mac SDL1 main
+#if defined(WAPI_SDL1) && defined(__APPLE__)
```

Comment:
No need to copy the entire function:
```
#if defined(...)
int SDL_main(.,.) {
#else
int main(...) {
#endif
  stuff();
  return 0;
}
```
Also, try linking with `-lSDL_main` or `-lSDL2_main` instead of renaming `main()`, that's kind of exactly what it's for.

Reply:
> No need to copy the entire function:
> 
> ```
> #if defined(...)
> int SDL_main(...) {
> #else
> int main(...) {
> #endif
>   stuff();
>   return 0;
> }
> ```
> 
> Also, try linking with `-lSDL_main` or `-lSDL2_main` instead of renaming `main()`, that's kind of exactly what it's for.

Thanks! Just tried your suggestion. Before I did, I looked at what sdl-config --libs outputs:
-L/usr/local/lib -lSDLmain -lSDL -Wl,-framework,Cocoa

I tried adding SDL_main to ldflags, but ld can't find SDL_main, and it's not in my 
/usr/local/lib directory. If you use the normal main it will compile but when executing It 
will never start, displaying some objc leaking errors, which AFAIK can only be solved by 
actually specifiying SDL_main in the main executable on Mac OS X.

I think this is what I found to solve the issue:
https://stackoverflow.com/questions/10965395/getting-sdl-to-work-on-osx-with-c-entry-point

Good catch on the better ifdef for main! I've changed it in the last commit. I think this is
the best solution. With further testing, I've noticed a regression in vsync using SDL2 in 
nightly, that does not happen in the master branch. Vsync in nightly uses different code,
which at least on PPC Mac OS X does not work, it randomly speeds everything.